provider/resources.go

```diff
@@ -3477,7 +3478,7 @@ func Provider() tfbridge.ProviderInfo {
 					res.Fields[azureLocation] = &tfbridge.SchemaInfo{
 						Name: azureLocation,
 						Default: &tfbridge.DefaultInfo{
-							From: defaultAzureLocation(p, rgRegionMap),
+							From: defaultAzureLocation(ctx, p, rgRegionMap),
```

Comment:
It seems like `From` should be changed to better support this type of behavior. A global `ctx` like this isn't really the best `ctx` here. `From` should be of type:

```go
func(context.Context, *tfbridge.PulumiResource) (interface{}, error)
```